src/components/form/form.component.js

```diff
@@ -0,0 +1,163 @@
+import React, {
+  useState, useRef, useEffect, useCallback
+} from 'react';
+import PropTypes from 'prop-types';
+import throttle from 'lodash/throttle';
+
+import ElementResize from '../../utils/helpers/element-resize';
+import OptionsHelper from '../../utils/helpers/options-helper';
+import FormSummary from './form-summary.component';
+import {
+  StyledForm,
+  StyledFormFooter,
+  StyledLeftButtons,
+  StyledRightButtons,
+  StyledSaveContainer
+} from './form.style';
+
+const SCROLL_THROTTLE = 100;
+
+const Form = ({
+  children,
+  saveButton,
+  leftSideButtons,
+  rightSideButtons,
+  errorCount,
+  warningCount,
+  onSubmit,
+  buttonAlignment,
+  stickyFooter,
+  showSummary,
+  ...rest
+}) => {
+  const [isFooterSticky, setIsFooterSticky] = useState(false);
+
+  const formRef = useRef();
+
+  const formFooterRef = useRef();
+
+  const stickyListenersAddedRef = useRef(false);
+
+  const checkStickyFooter = useCallback(throttle(() => {
+    const formHeight = (formRef.current.offsetTop + formRef.current.offsetHeight) - window.pageYOffset;
+    const footerHeight = formFooterRef.current.offsetHeight;
+
+    if (formHeight - (footerHeight / 2) > window.innerHeight) {
+      setIsFooterSticky(true);
+    } else if (formHeight < window.innerHeight) {
+      setIsFooterSticky(false);
+    }
+  }, SCROLL_THROTTLE), []);
+
+  const addStickyFooterListeners = useCallback(() => {
+    ElementResize.addListener(formRef.current, checkStickyFooter);
+    window.addEventListener('resize', checkStickyFooter, true);
+    window.addEventListener('scroll', checkStickyFooter, true);
+  }, [checkStickyFooter]);
+
+  const removeStickyFooterListeners = useCallback(() => {
+    ElementResize.removeListener(formRef.current, checkStickyFooter);
+    window.removeEventListener('resize', checkStickyFooter, true);
+    window.removeEventListener('scroll', checkStickyFooter, true);
+  }, [checkStickyFooter]);
+
+  useEffect(() => {
+    if (stickyFooter && !stickyListenersAddedRef.current) {
```

Comment:
I see that the cleanup of this effect will call `removeStickyFooterListeners` but won't reset the guard `stickyListenersAddedRef`.

> This is why React also cleans up effects from the previous render before running the effects next time
> https://reactjs.org/docs/hooks-effect.html

So if the component is rendered again, the listener will be removed and won't be added again.